game-core/src/test/java/games/strategy/triplea/delegate/battle/casualty/AaCasualtySelectorTest.java

```diff
@@ -0,0 +1,432 @@
+package games.strategy.triplea.delegate.battle.casualty;
+
+import static games.strategy.triplea.Constants.UNIT_ATTACHMENT_NAME;
+import static games.strategy.triplea.delegate.MockDelegateBridge.whenGetRandom;
+import static games.strategy.triplea.delegate.MockDelegateBridge.withValues;
+import static games.strategy.triplea.delegate.battle.steps.MockGameData.givenGameData;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.empty;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.description;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.GamePlayer;
+import games.strategy.engine.data.Territory;
+import games.strategy.engine.data.Unit;
+import games.strategy.engine.data.UnitType;
+import games.strategy.engine.delegate.IDelegateBridge;
+import games.strategy.triplea.attachments.UnitAttachment;
+import games.strategy.triplea.delegate.DiceRoll;
+import games.strategy.triplea.delegate.battle.BattleState;
+import games.strategy.triplea.delegate.data.CasualtyDetails;
+import games.strategy.triplea.delegate.power.calculator.CombatValue;
+import games.strategy.triplea.delegate.power.calculator.CombatValueBuilder;
+import java.util.List;
+import java.util.UUID;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class AaCasualtySelectorTest {
+
+  @Mock GamePlayer hitPlayer;
+  @Mock GamePlayer aaPlayer;
+  private UnitType aaUnitType;
+  private UnitType damageableAaUnitType;
+  private UnitType planeUnitType;
+  private UnitType planeMultiHpUnitType;
+  private GameData gameData;
+
+  private List<Unit> givenAaUnits(final int quantity) {
+    return aaUnitType.create(quantity, aaPlayer, true);
+  }
+
+  private List<Unit> givenDamageableAaUnits(final int quantity) {
+    return damageableAaUnitType.create(quantity, aaPlayer, true);
+  }
+
+  private List<Unit> givenPlaneUnits(final int quantity) {
```

Comment:
Consider inlining this method, potentially needless indirection. The one liner is more self explanatory than the method name. Seeing `givenPlaneUnits(10)`, it's not clear what the 10 is. The boolean parameter on the `create` API is needless and we should create an overload on the `create` method that has a default 'true' on it so the API can then make a lot of sense and be just "create units, count, player".

Reply:
Then perhaps:
```
  public List<Unit> createTemp(final int quantity, final GamePlayer owner) {
    return create(quantity, owner);
  }
```

With that it's clear you are creating a 'temp' unit. (What a temp unit is, is another question altogether, but at least the magic boolean is removed).

Reply:
Let me do that in another PR.  There's a lot of places that use the temp = true and it would be nice to keep them all unified.